node/src/database/db_migrations.rs

```diff
@@ -74,353 +158,684 @@ impl DbMigratorReal {
+        if current_state.lt(target_version) {
+            false
+        } else if current_state.eq(target_version) {
+            true
         } else {
-            CURRENT_SCHEMA_VERSION
-        };
-        (
-            first,
-            Self::identification_of_versions(first.version_compatibility(), second_by_its_name),
-        )
+            panic!("Nonsense: the given target is lower than the version that is considered mismatched")
+        }
     }
 
-    fn check_the_number_of_those_remaining(
-        mismatched_schema: &str,
-        count: usize,
-    ) -> Result<(), String> {
-        match count {
-            0 => Err(format!("Database claims to be more advanced ({}) than the version {} which is the latest released.", mismatched_schema, CURRENT_SCHEMA_VERSION)),
-            _ => Ok(())
+    fn select_updates_to_process<'a>(
+        mismatched_schema: usize,
+        list_of_updates: &'a [&'a (dyn DatabaseMigration + 'a)],
+    ) -> Vec<&'a (dyn DatabaseMigration + 'a)> {
+        let updates_to_process = list_of_updates
+            .iter()
+            .skip_while(|entry| entry.old_version().ne(&mismatched_schema))
+            .map(Self::deref)
+            .collect::<Vec<&(dyn DatabaseMigration + 'a)>>();
+        let _ = Self::check_out_quantity_of_those_remaining(
+            mismatched_schema,
+            updates_to_process.len(),
+        );
+        updates_to_process
+    }
+
+    fn deref<'a>(value: &'a &dyn DatabaseMigration) -> &'a dyn DatabaseMigration {
+        *value
+    }
+
+    fn process_items_with_dirty_references<'a>(
+        first: Option<&'a &dyn DatabaseMigration>,
+        second: Option<&&&dyn DatabaseMigration>,
+    ) -> (&'a dyn DatabaseMigration, String) {
+        let first = *first.expect_v("migration record");
+        let identity_of_the_second = Self::identify_the_next_record(second, first);
+        (first, identity_of_the_second)
+    }
+
+    fn identify_the_next_record(
+        subject: Option<&&&dyn DatabaseMigration>,
+        current_record: &dyn DatabaseMigration,
+    ) -> String {
+        if let Some(next_higher) = subject {
+            next_higher.old_version().to_string()
+        } else {
+            (current_record.old_version() + 1).to_string()
         }
     }
 
-    fn prepare_mailing_of_bad_news(
-        &self,
-        versions: &str,
-        error: rusqlite::Error,
-    ) -> Result<(), String> {
+    fn check_out_quantity_of_those_remaining(mismatched_schema: usize, count: usize) {
+        if count == 0 {
+            panic!("Database claims to be more advanced ({}) than the version {} which is the latest released.", mismatched_schema, CURRENT_SCHEMA_VERSION)
+        }
+    }
+
+    fn dispatch_bad_news(&self, versions: &str, error: rusqlite::Error) -> Result<(), String> {
         let error_message = format!("Updating database {} failed: {:?}", versions, error);
         warning!(self.logger, "{}", &error_message);
```

Comment:
This probably should be an error!(). Or nothing, if you're concerned about repetition in the log file.